### PR TITLE
Implemented support for server running over TRAMP

### LINKED ISSUE #197: Run lsp-mode in Tramp buffers
I've been poking around a bit at `lsp-mode` to see what it would take to [integrate it with Tramp](https://www.gnu.org/software/emacs/manual/html_node/tramp/Remote-processes.html). Blocker right now is that the call to `(make-process)` doesn't have any Tramp hooks.

Other packages use `start-file-process` to pick up Tramp support, but that function doesn't support stderr-separation. Might need to get more creative for lsp-mode.

### PR DESCRIPTION
**Pull request by yyoncho**
Fixes #197

- refer to readme for implementation details and how to use
- more idomatic emacs lisp - (replacing if+let with if-let idiomatic clojure)
  destructoring

### CODE REVIEW

**Comment by yyoncho** [comment]
Merging the PR based on the informal review/testing in: 
https://github.com/yyoncho/lsp-mode/commit/c586861de17f5dc3b374458ffeca02fd8e24c1c1#commitcomment-31872770 by @Kypert

File: lsp-mode.el [hunk 1]
```diff
@@ -2967,35 +2949,54 @@ SLEEP-INTERVAL is the sleep interval between each retry."
         (file-error (setq success t))))
     port))
 
-(defun lsp-tcp-connection (command-fn)
+(defun lsp-tcp-connection (command)
```
Comment (kenkangxgwe) [inline lsp-mode.el]: Sorry, but why was this renamed to `command`

Comment (yyoncho) [reply lsp-mode.el]: It looks like a leftover change, will be fixed.

Comment (yyoncho) [reply lsp-mode.el]: Fixed with 03e727718e6da93a2344392545534102de715c5c
